sqlmesh/core/state_sync/engine_adapter.py

```diff
@@ -397,7 +538,9 @@ def _get_snapshots_with_same_version(
             query = query.lock(copy=False)
 
         snapshot_rows = self.engine_adapter.fetchall(query, ignore_unsupported_errors=True)
-        return [Snapshot(**json.loads(row[0])) for row in snapshot_rows]
+        return self._snapshot_cache.get_cached_snapshots(
```

Comment:
The name is very misleading here.

Reply:
Also it seems like if snapshots were deleted from the state we'd still return them here if they happened to be in the cache.